File: src/main/java/com/example/installcertatruntime/cert/SSLClasspathTrustStoreLoader.java
package com.example.installcertatruntime.cert;

import java.io.InputStream;
import java.security.KeyStore;

import javax.net.ssl.SSLContext;
import javax.net.ssl.TrustManager;
import javax.net.ssl.TrustManagerFactory;

public class SSLClasspathTrustStoreLoader {

    public static void setTrustStore(String trustStore, String password) throws Exception {
        TrustManagerFactory trustManagerFactory = TrustManagerFactory.getInstance("X509");
        KeyStore keystore = KeyStore.getInstance(KeyStore.getDefaultType());
        InputStream keystoreStream = SSLClasspathTrustStoreLoader.class.getResourceAsStream(trustStore);
        keystore.load(keystoreStream, password.toCharArray());
        trustManagerFactory.init(keystore);
        TrustManager[] trustManagers = trustManagerFactory.getTrustManagers();
        SSLContext sslContext = SSLContext.getInstance("SSL");
        sslContext.init(null, trustManagers, null);
        SSLContext.setDefault(sslContext);
    }

}

File: src/main/kotlin/com/example/installcertatruntime/cert/SSLClasspathTrustStoreLoader.kt
package com.example.installcertatruntime.cert

import java.security.KeyStore
import javax.net.ssl.SSLContext
import javax.net.ssl.TrustManagerFactory

object SSLClasspathTrustStoreLoader {

    @Throws(Exception::class)
    fun setTrustStore(trustStore: String, password: String) {
        val trustManagerFactory = TrustManagerFactory.getInstance("X509")
        val keystore = KeyStore.getInstance(KeyStore.getDefaultType())
        val keystoreStream = SSLClasspathTrustStoreLoader::class.java.getResourceAsStream(trustStore)
        keystore.load(keystoreStream, password.toCharArray())
        trustManagerFactory.init(keystore)
        val trustManagers = trustManagerFactory.trustManagers
        val sslContext = SSLContext.getInstance("SSL")
        sslContext.init(null, trustManagers, null)
        SSLContext.setDefault(sslContext)
    }

}